tests/unit_tests/test_analytics/test_sentry.py

```diff
@@ -0,0 +1,593 @@
+import os
+from unittest import mock
+
+import pytest
+import wandb
+import wandb.analytics
+import wandb.env
+from sentry_relay import MetricRelayServer
+from sentry_sdk.utils import sentry_sdk
+from tests.unit_tests.test_analytics.sentry_relay import SentryResponse
+
+SENTRY_DSN_FORMAT = "http://{key}@127.0.0.1:{port}/{project}"
+
+
+@pytest.fixture(scope="module")
+def relay():
+    relay_server = MetricRelayServer()
+    relay_server.start()
+
+    yield relay_server
+
+    relay_server.stop()
+
+
+def test_wandb_sentry_does_not_interfer_with_global_sentry_sdk(
+    relay: MetricRelayServer,
+):
+    """
+    Test that wandb sentry initialization does not interfere with global sentry_sdk.
+    """
+    other_sentry_public_key = "OTHER_SENTRY_PUBLIC_KEY"
+    other_sentry_project_id = "654321"
+    wandb_sentry_public_key = "WANDB_SENTRY_PUBLIC_KEY"
+    wandb_sentry_project_id = "123456"
```

Comment:
probably don't need these as are only used once,
if you need to make a distinction, could just leave a comment